File: app/src/main/java/com/lannix/netfilmdata/fragment/expand_page/ExpandPage1.java
package com.lannix.netfilmdata.fragment.expand_page;

import android.content.Context;
import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;

import com.lannix.netfilmdata.R;
import com.lannix.netfilmdata.fragment.MainActivityInterface;
import com.lannix.netfilmdata.stuff.OnBackPressedInterface;

public class ExpandPage1 extends Fragment implements OnBackPressedInterface {

    private Context context;
    private MainActivityInterface mainActivityInterface;

    @Nullable
    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        View view = inflater.inflate(R.layout.fragment_page_1, container, false);
        return view;
    }

    @Override
    public void onAttach(Context context) {
        super.onAttach(context);
        this.context = context;
        if (context instanceof MainActivityInterface) {
            mainActivityInterface = (MainActivityInterface) context;
        } else {
            throw new RuntimeException(context.toString()
                    + " must implement OnFragmentInteractionListener");
        }
    }

    @Override
    public void onBackPressedAction() {
        getActivity().onBackPressed();
    }
}

File: app/src/main/java/com/lannix/netfilmdata/fragment/expand_page/ExpandPage1.kt
package com.lannix.netfilmdata.fragment.expand_page

import android.content.Context
import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.fragment.app.Fragment

import com.lannix.netfilmdata.R
import com.lannix.netfilmdata.fragment.MainActivityInterface
import com.lannix.netfilmdata.stuff.OnBackPressedInterface

class ExpandPage1 : Fragment(), OnBackPressedInterface {

    private var context: Context? = null
    private var mainActivityInterface: MainActivityInterface? = null

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        return inflater.inflate(R.layout.fragment_page_1, container, false)
    }

    override fun onAttach(context: Context?) {
        super.onAttach(context)
        this.context = context
        if (context is MainActivityInterface) {
            mainActivityInterface = context
        } else {
            throw RuntimeException(context!!.toString() + " must implement OnFragmentInteractionListener")
        }
    }

    override fun onBackPressedAction() {
        activity!!.onBackPressed()
    }
}
